### PR TITLE
Update gcta version

### PR DESCRIPTION
**Pull request by gmauro**
Update gcta to version 1.94.1

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/gcta/meta.yaml [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
 test:
   commands:
     - gcta64 | grep "Analysis started"
 
 about:
-  home: "https://cnsgenomics.com/software/gcta/"
-  license: MIT
+  home: "https://yanglab.westlake.edu.cn/software/gcta"
+  license: GPLv3
```
Comment (bgruening) [inline recipes/gcta/meta.yaml:23]: Can you please use SPDX identifiers, eg. https://spdx.org/licenses/GPL-3.0-only.html

File: recipes/gcta/meta.yaml [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
   name: "{{ name|lower }}"
   version: "{{ version }}"
 
-source:
-  - url: https://cnsgenomics.com/software/{{ name }}/bin/{{ name }}_{{ version }}.zip # [linux]
-    sha256: 44354f4909666b760ab3cf538dc143ee3feff5540c2ad237a098766ff9829374 # [linux]
-  - url: https://cnsgenomics.com/software/{{ name }}/bin/{{ name }}_{{ version }}_mac.zip # [osx]
-    sha256: b92f4ac51e5e5cb04c8eb93d07605efb9f8847ec0afcb8df2717200f995b6fe7 # [osx]
+source: https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip
+  - url: https://yanglab.westlake.edu.cn/software/{{ name }}/bin/{{ name }}-{{ version }}-linux-kernel-3-x86_64.zip # [linux]
```
Comment (bgruening) [inline recipes/gcta/meta.yaml:9]: the linter bails out with:

````
ruamel.yaml.scanner.ScannerError: mapping values are not allowed here
  in "<unicode string>", line 9, column 8:
      - url: https://yanglab.westlake.edu.c/ ... 
           ^ (line: 9)
```

File: recipes/gcta/meta.yaml (hunk 1 above)
Comment (gmauro) [reply recipes/gcta/meta.yaml]: Yes, sure

File: recipes/gcta/meta.yaml (hunk 2 above)
Comment (gmauro) [reply recipes/gcta/meta.yaml:9]: I guess I got it. Let's see

File: recipes/gcta/meta.yaml [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+    sha256: 51c2a5c622cb7ab8d246252abece9d28bf6e41c0e54a0b820a0871695a269441 # [osx]
 
 build:
-  number: 1
+  number: 0
 
 test:
   commands:
-    - gcta64 | grep "Analysis started"
+    - gcta-1.94.1 | grep "Analysis started"
```
Comment (bgruening) [inline recipes/gcta/meta.yaml:19]: Would it make sense to create symlinks to a gcta binary, or even the gcta64 binary?

Comment (gmauro) [reply recipes/gcta/meta.yaml]: Yes, definitely! Thanks for pointing it out!

**Review by bgruening — APPROVED** [review]

**Pull request merged**